ssl/ssl_x509.cc

```diff
@@ -160,13 +160,15 @@ BSSL_NAMESPACE_BEGIN
 // installed. Calling an X509-based method on an |ssl| with a different method
 // will likely misbehave and possibly crash or leak memory.
 static void check_ssl_x509_method(const SSL *ssl) {
-  assert(ssl == NULL || ssl->ctx->x509_method == &ssl_crypto_x509_method);
+  assert(ssl == nullptr || (ssl != nullptr && ssl->ctx != nullptr &&
+                            ssl->ctx->x509_method == &ssl_crypto_x509_method));
```

Comment:
The `ssl != nullptr` is not needed. 

Should this fail when `ssl->ctx == nullptr`? Should this be:
```
  assert(ssl == nullptr || 
            ssl->ctx == nullptr ||
                ssl->ctx->x509_method == &ssl_crypto_x509_method));
```
?

Reply:
I'm not 100% sure and I'll just remove this change since this is getting out of scope of this particular PR. Asserts against null checks in libssl are definitely nice, but we should open this as another task and resolve these all at once.